src/main/java/org/jabref/logic/importer/fetcher/CitationRelationFetcher.java

```diff
@@ -0,0 +1,137 @@
+package org.jabref.logic.importer.fetcher;
+
+import java.io.*;
+import java.net.SocketException;
+import java.net.URL;
+import java.net.UnknownHostException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javafx.beans.property.DoubleProperty;
+import javafx.beans.property.SimpleDoubleProperty;
+import kong.unirest.json.JSONArray;
+import kong.unirest.json.JSONException;
+
+import org.jabref.logic.importer.EntryBasedFetcher;
+import org.jabref.logic.importer.FetcherException;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.preferences.JabRefPreferences;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class to fetch for an articles citation relations on opencitations.net's API
+ */
+public class CitationRelationFetcher implements EntryBasedFetcher {
```

Comment:
The `EntryBasedFetcher` interface describes fetcher that fetch bibliographic information for a given entry. The use case is that users have a partially filled entry (say only having a title and author info) and want to get other information like the doi, isbn, etc. 

While it is possible to broaden the scope of `EntryBasedFetcher` to also include citation information, I would prefer to create a new fetcher interface for this. For example:
```
interface CitationFetcher:
   List<BibEntry> searchCitedBy(BibEntry entry);
   List<BibEntry> searchCiting(BibEntry entry);
```
Then also introduce an abstract helper class  `CitationParserBasedFetcher` that uses a common parser for these search methods (similar to the other "parser based" fetcher). Then the current `CitationRelationFetcher` can derive from this new helper class. Maybe also rename `CitationRelationFetcher` to `OpenCitationFetcher` so that it adheres to the current naming conventions.

This has also the advantage, that one could easily add further citation fetcher in the future.

Reply:
We created the interface CitationFetcher as you suggested, but we're not shure what advantages the abstract helper class whould have. Could you clarify the benefit?

Reply:
The purpose of the helper class is mainly to make the design a bit cleaner and prepare the implementation of other citation fetcher in the future. 

If you still have time, then this is something to work on; but it's not super important.

Reply:
Done! If the implementation differs greatly from your intended solution, feel free to suggest changes.